metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -2024,13 +2029,8 @@ public CompletableFuture<ElectLeadersResponseData> electLeaders(
     public CompletableFuture<FinalizedControllerFeatures> finalizedFeatures(
         ControllerRequestContext context
     ) {
-        // It's possible that we call ApiVersionRequest before consuming the log since ApiVersionRequest is sent when
-        // initialize NetworkClient, we should not return an error since it would stop the NetworkClient from working correctly.
-        if (lastCommittedOffset == -1) {
```

Comment:
Was this a bug previously? If another node sent an ApiVersionRequest prior to loading the log, we would have not returned any MetadataVersion (whereas with this patch, we would).